Generators/src/GeneratorHepMC.cxx

```diff
@@ -159,7 +159,7 @@ Bool_t GeneratorHepMC::generateEvent()
     tries++;
   } while (tries < max_tries);
 
-  LOG(fatal) << "HepMC event gen failed (Does the file/stream have enough events)?";
+  LOG(error) << "HepMC event gen failed (Does the file/stream have enough events)?";
```

Comment:
This change provides the caller of the generateEvent method the opportunity to decide what happens in case the event can not be properly read from the HepMC file.

In the case I am implementing the reaction is, instead of stopping the process, to generate a new HepMC file and to restart generateEvent with the new file.

Reply:
ok understood